src/VssDatabase.cpp

```diff
@@ -701,8 +753,22 @@ void VssDatabase::setSignal(WsChannel& channel,
     throw genException(msg);
   }
   checkSetPermission(channel, path);
+  
+  jsoncons::json setValues;
+
   rwMutex_.lock();
-  jsoncons::json setValues = getPathForSet(path, valueJson);
+  try {
+      setValues = getPathForSet(path, valueJson);
+  }
+  catch( noPathFoundonTree& e) {
+    rwMutex_.unlock();
+    throw e;
+  }
+  catch ( genException &e) {
+    logger_->Log(LogLevel::ERROR, "Exception VssDatabase::setSignal: " + string(e.what()));
+    rwMutex_.unlock();
```

Comment:
using https://en.cppreference.com/w/cpp/thread/lock_guard will solve the problem better, but yes, we can refactor it later